src/types/redis_zset.cc

```diff
@@ -707,6 +707,102 @@ rocksdb::Status ZSet::InterStore(const Slice &dst, const std::vector<KeyWeight>
   return rocksdb::Status::OK();
 }
 
+rocksdb::Status ZSet::Inter(const std::vector<KeyWeight> &keys_weights, AggregateMethod aggregate_method,
+                            MemberScores *mscores) {
+  if (!mscores->empty()) {
+    mscores->clear();
+  }
+  std::map<std::string, double> dst_zset;
+  std::map<std::string, size_t> member_counters;
+  std::vector<MemberScore> target_mscores;
+  uint64_t target_size = 0;
+  RangeScoreSpec spec;
+  auto s = RangeByScore(keys_weights[0].key, spec, &target_mscores, &target_size);
+  if (!s.ok() || target_mscores.empty()) return s;
+
+  for (const auto &ms : target_mscores) {
+    double score = ms.score * keys_weights[0].weight;
+    if (std::isnan(score)) score = 0;
+    dst_zset[ms.member] = score;
+    member_counters[ms.member] = 1;
+  }
+
+  for (size_t i = 1; i < keys_weights.size(); i++) {
+    s = RangeByScore(keys_weights[i].key, spec, &target_mscores, &target_size);
+    if (!s.ok() || target_mscores.empty()) return s;
+
+    for (const auto &ms : target_mscores) {
+      if (dst_zset.find(ms.member) == dst_zset.end()) continue;
+      member_counters[ms.member]++;
+      double score = ms.score * keys_weights[i].weight;
+      if (std::isnan(score)) score = 0;
+      switch (aggregate_method) {
+        case kAggregateSum:
+          dst_zset[ms.member] += score;
+          if (std::isnan(dst_zset[ms.member])) {
+            dst_zset[ms.member] = 0;
+          }
+          break;
+        case kAggregateMin:
+          if (dst_zset[ms.member] > score) {
+            dst_zset[ms.member] = score;
+          }
+          break;
+        case kAggregateMax:
+          if (dst_zset[ms.member] < score) {
+            dst_zset[ms.member] = score;
+          }
+          break;
+      }
+    }
+  }
+
+  for (const auto &iter : dst_zset) {
+    if (member_counters[iter.first] != keys_weights.size()) continue;
+    mscores->emplace_back(MemberScore{iter.first, iter.second});
+  }
+  return rocksdb::Status::OK();
+}
+
+rocksdb::Status ZSet::InterCard(const std::vector<KeyWeight> &keys_weights, uint64_t limit, uint64_t *saved_cnt) {
```

Comment:
To be completely sure, I'd also check if the `keys_weights` is empty and make an early return because later we use the `keys_weights[0].key` value and it will lead to a runtime error. The same could be applied to the `Inter` function.

Reply:
ok，i will check it